version_with_class.py

```diff
@@ -0,0 +1,191 @@
+import os
+import tempfile
+from functools import wraps
+import random
+import subprocess
+
+from typing import Dict, Tuple, List, Generator, Callable, List
+from iib.exceptions import IIBError, AddressAlreadyInUse
+
+
+class Config:
+    iib_opm_port_ranges: Dict[str, Tuple[int, int]] = {
+        "opm_port": (50051, 50151),
+        "opm_ppof_port": (50151, 50251),
+    }
+    iib_opm_pprof_lock_required_versions = ["1.40.0"]
+
+class Opm:
+    opm_version = "1.40.0"
+
+def get_worker_config():
+    return Config()
+
+class PortFileLock:
+    def __init__(self, purpose: str, port: int, lock_on_init: bool = False):
+        self.purpose = purpose
+        self.port = port
+        self.locked = False
+        self.filename = os.path.join(
+            tempfile.gettempdir(),
+            f'iib_{purpose}_{port}.lock',
+        )
+        if lock_on_init:
+            self.lock_acquire()
+
+    def lock_acquire(self):
+        os.open(self.filename, os.O_CREAT | os.O_EXCL | os.O_RDWR)
+        self.locked = True
+        # log.debug()
+
+    def unlock(self):
+        os.remove(self.filename)
+        self.locked = False
+        # log.debug(f'Port {port} is already in use. Checking next one...')
+
+
+def port_locks_generator(
+    port_stacks: List[List[int]],
+    ports_descriptions: List[str]
+) -> Generator[List[PortFileLock], None, None]:
+    """
+    Generate port for opm service from range set in IIB config.
+
+    :raises: IIBError when all ports were already taken
+    """
+    num_of_attempts = len(port_stacks)
+
+    # create port_lock_here
+    while port_stacks:
+        port_numbers = port_stacks.pop(0)
+        new_locks = [
+            PortFileLock(
+                purpose=description,
+                port=port_numbers[port_position],
+            )
+            for port_position, description in enumerate(ports_descriptions)
+        ]
+        yield new_locks
+
+    # The port stack is empty - we tried out all ports from allowed range
+    # therefore we will raise and IIB error
+    err_msg = f'No free port has been found after {num_of_attempts} attempts.'
+    print(err_msg)
+    raise Exception(err_msg)
+
+def get_port_stacks() -> Tuple[List[List[int]], List[str]]:
+    conf = get_worker_config()
+    ports_description = list(conf.iib_opm_port_ranges)
+    if Opm.opm_version in conf.iib_opm_pprof_lock_required_versions:
```

Comment:
This might be simple `if Opm.opm_version >= 1.29.0` (however it would be great to have the version in config)

Reply:
@lipoja please check